samza-core/src/test/java/org/apache/samza/clustermanager/TestContainerAllocatorWithHostAffinity.java

```diff
@@ -350,6 +358,146 @@ public void testExpiredRequestsAreCancelled() throws Exception {
     Assert.assertEquals(clusterResourceManager.cancelledRequests.size(), 3);
   }
 
+  @Test
+  public void testRequestAllocationOnPreferredHostWithRunStreamProcessor() throws Exception {
+    ClusterResourceManager.Callback mockCPM = mock(MockClusterResourceManagerCallback.class);
+    // Mock the callback from ClusterManager to add resources to the allocator
+    doAnswer((InvocationOnMock invocation) -> {
+        SamzaResource resource = (SamzaResource) invocation.getArgumentAt(0, List.class).get(0);
+        spyAllocator.addResource(resource);
+        return null;
+      }).when(mockCPM).onResourcesAvailable(anyList());
+
+    spyAllocator = Mockito.spy(
+        new AbstractContainerAllocator(new MockClusterResourceManager(mockCPM, state), config, state,
+            getClass().getClassLoader(), true, Optional.empty()));
+
+    // Request Resources
+    spyAllocator.requestResources(new HashMap<String, String>() {
+      {
+        put("0", "abc");
```

Comment:
Do "abc" and "xyz" correspond to hostnames?

Reply:
yes

Reply:
Can we just use "hostname-0" and "hostname-1" in that case?